engine/gamesys/src/gamesys/components/comp_sound.cpp

```diff
@@ -430,7 +449,17 @@ namespace dmGameSystem
                     dmSound::SetLooping(entry.m_SoundInstance, sound->m_Looping, (sound->m_Looping && !sound->m_Loopcount) ? -1 : sound->m_Loopcount ); // loopcounter semantics differ a bit from loopcount. If -1, it means loopforever, otherwise it contains the # of loops remaining.
 
                     entry.m_Listener = params.m_Message->m_Sender;
-                    entry.m_LuaCallback = params.m_Message->m_UserData2;
+                    uintptr_t callback = params.m_Message->m_UserData2;
+                    if (callback == UINTPTR_MAX)
+                    {
+                        // UINTPTR_MAX used as default for `sound.play()` in `script_sound.cpp`
+                        // to make distinguish between function call and `play_sound` message
+                        // if it's a function call without callback (UINTPTR_MAX) then event
+                        // shouldn't be dispatched
+                        callback = callback == UINTPTR_MAX ? 0x0 : callback;
+                        entry.m_ShouldDispatchEvents = 0;
```

Comment:
Before the last changes, you checked if the m_LuaCallback was set.
Doesn't that mean you should check if it's true here as well then?

E.g. 
```
if (callback)
    entry.m_ShouldDispatchEvents = 1;
```

Reply:
It doesn't matter if it's set or not with current apprach. 
we have 4 situations:
1. sound.play(callback) -- Should Dispatch
2. sound.play() -- Should't and UINTPTR_MAX as flag
3. msg.post() --Should't and  play_id is INVALID_PLAY_ID
4. msg.post(play_id) -- Should Dispatch

this line exclude #3
>entry.m_ShouldDispatchEvents = entry.m_PlayId != dmSound::INVALID_PLAY_ID;

this condition exclude #2
>if (callback == UINTPTR_MAX)

1 and 4 are fine to call, if m_LuaCallback is 0x0 - msg will be posted #4, otherwise #1

Reply:
RIght so with sound.play() we always have a playid != invalid.

Reply:
yes